### DISCUSSION TITLE
使用addEdge遇到一种情况，该使用那种方式处理

### DISCUSSION THREAD

**Discussion by tunshiyu** [antvis/X6#2588] [Q&A]
场景： 监控平台 => keepalive请求拿到status:2  => 通过addEdge绘制图形 => edge为2对应的样式 => keepalive请求拿到status:4，状态变更 => useeffect检测到status改变执行addEdge => 新的edge覆盖了原来的edge重影了

我目前想到addEdge返回的id，在useeffect里当status改变时，graph.removeEdge掉对应的线。有没有graph.rePaint这种概念的API，可以在某些情况下让graph恢复到最开始的状态

**Comment by zmcode** [comment]
哪怕恢复到以前的状态也是需要重新绘制的, 不可避免的问题
像你这个对单个边样式变化, 使用update比较好, 如果需要对整个画布进行重绘, 建议在edge改变的时候, 存一份JSON数据到localstorage, 想变回来, 拿localstorage数据重新绘制
